pkg/ingester/owned_series_test.go

```diff
@@ -650,123 +650,860 @@ func TestOwnedSeriesService(t *testing.T) {
+
+			c.ing = setupIngesterWithOverridesAndPartitionRing(t, c.cfg, c.overrides, c.partitionsRing, "")
+
+			c.buf = &concurrency.SyncBuffer{}
+
+			// We also don't start this service, and will only call its methods.
+			c.ownedSeries = newOwnedSeriesService(
+				10*time.Minute,
+				newOwnedSeriesPartitionRingStrategy(c.ing.ingestPartitionID, c.partitionsRing, c.ing.limits.IngestionPartitionsTenantShardSize),
+				log.NewLogfmtLogger(c.buf),
+				nil,
+				c.ing.limiter.maxSeriesPerUser,
+				c.ing.getTSDBUsers,
+				c.ing.getTSDB,
+			)
+
+			tc.testFunc(t, &c, tc.limits)
+		})
+	}
+}
+
+func TestOwnedSeriesIngesterRingStrategyRingChanged(t *testing.T) {
+	kvStore, closer := consul.NewInMemoryClient(ring.GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	wkv := &watchingKV{Client: kvStore}
+
+	rc := ring.Config{}
+	flagext.DefaultValues(&rc)
+
+	// Configure ring
+	rc.KVStore.Mock = wkv
+	rc.HeartbeatTimeout = 1 * time.Minute
+	rc.ReplicationFactor = 3
+	rc.ZoneAwarenessEnabled = true
+
+	rng := createAndStartRing(t, rc)
+
+	const instanceID1 = "first ingester"
+	const instanceID2 = "second instance"
+
+	ringStrategy := newOwnedSeriesIngesterRingStrategy(instanceID1, rng, nil)
+
+	updateRingAndWaitForWatcherToReadUpdate(t, wkv, func(desc *ring.Desc) {
+		desc.AddIngester(instanceID1, "localhost:11111", "zone", []uint32{1, 2, 3}, ring.ACTIVE, time.Now())
+	})
+
+	// First call should indicate ring change.
+	t.Run("first call always reports change", func(t *testing.T) {
+		changed, err := ringStrategy.checkRingForChanges()
+		require.NoError(t, err)
+		require.True(t, changed)
+	})
+
+	t.Run("second call (without ring change) reports no change", func(t *testing.T) {
+		changed, err := ringStrategy.checkRingForChanges()
+		require.NoError(t, err)
+		require.False(t, changed)
+	})
+
+	t.Run("new instance added", func(t *testing.T) {
+		updateRingAndWaitForWatcherToReadUpdate(t, wkv, func(desc *ring.Desc) {
+			desc.AddIngester(instanceID2, "localhost:22222", "zone", []uint32{4, 5, 6}, ring.ACTIVE, time.Now())
+		})
+
+		changed, err := ringStrategy.checkRingForChanges()
+		require.NoError(t, err)
+		require.True(t, changed)
+	})
+
+	t.Run("change of state is not interesting", func(t *testing.T) {
+		updateRingAndWaitForWatcherToReadUpdate(t, wkv, func(desc *ring.Desc) {
+			desc.AddIngester(instanceID2, "localhost:22222", "zone", []uint32{4, 5, 6}, ring.LEAVING, time.Now())
+		})
+
+		// Change of state is not interesting.
+		changed, err := ringStrategy.checkRingForChanges()
+		require.NoError(t, err)
+		require.False(t, changed)
+	})
+
+	t.Run("removal of instance", func(t *testing.T) {
+		updateRingAndWaitForWatcherToReadUpdate(t, wkv, func(desc *ring.Desc) {
+			desc.RemoveIngester(instanceID2)
+		})
+
+		changed, err := ringStrategy.checkRingForChanges()
+		require.NoError(t, err)
+		require.True(t, changed)
+	})
+}
+
+func TestOwnedSeriesPartitionsRingStrategyRingChanged(t *testing.T) {
```

Comment:
This is `TestOwnedSeriesIngesterRingStrategyRingChanged` adapted for partitions. Some test cases are different.